### PR TITLE
Remy

### PR DESCRIPTION
**Pull request by CosmoRemyG**
Implementation of a socket client.
Kite-common 0.1.2 not published yet.

### CODE REVIEW

File: wdmgmt/WebDriverUtility.js [hunk 1]
```diff
@@ -2,34 +2,32 @@ const firefox = require('selenium-webdriver/firefox');
...
 const getFirefoxOptions = function(browser) {
   let firefoxOptions;
-  let path = process.env.KITE_HOME.split("\\").join('/');
-  let profile = path + "/third_party/";
-  if (profile != undefined) {
+  let kiteHome = process.env.KITE_HOME;
+  if (kiteHome != undefined) {
+    let path = kiteHome.split("\\").join('/');
+    let profile = path + "/third_party/";
     switch(browser.platform) {
```
Comment (ManuCosmo) [inline wdmgmt/WebDriverUtility.js]: better keep the caps and make is case insensitive like:
`switch(browser.platform.toUpperCase()) {`

File: package.json [hunk 2]
```diff
@@ -1,34 +1,35 @@
...
     "/"
   ],
-  "_resolved": "https://registry.npmjs.org/kite-common/-/kite-common-0.0.8.tgz",
-  "_shasum": "486e8bea1e462b964138a1179128fa9f1dce7552",
-  "_spec": "kite-common@0.0.8",
-  "_where": "C:\\newGitHub\\KITE-Sample-Tests\\KITE-Mediasoup-Test\\js",
+  "_resolved": "https://registry.npmjs.org/kite-common/-/kite-common-0.1.1.tgz",
+  "_shasum": "ab5245d42e76661fc4cfc2c5b2f3c5c7bb901b9d",
+  "_spec": "kite-common@0.1.1",
+  "_where": "C:\\newGitHub\\KITE-Sample-Tests\\KITE-Janus-Test\\js",
```
Comment (namvuCosmo) [inline package.json]: probably not commit path like this

File: wdmgmt/WebDriverFactory.js [hunk 3]
```diff
@@ -1,13 +1,40 @@
...
+  if (platform.includes('mac') || platform.includes('darwin')) {
+    return 'Mac';
+  }
+  if (platform.includes('nux')) {
+    return 'Linux';
+  }
+  if (platform.includes('nix') || platform.includes('aix') 
+    || platform.includes('freebsd') || platform.includes('openbsd')
+    || platform.includes('sunos')) {
+    return 'Unix';
```
Comment (namvuCosmo) [inline wdmgmt/WebDriverFactory.js]: What is the use of having a platform "Unix"

File: wdmgmt/WebDriverUtility.js [hunk 4]
```diff
@@ -2,10 +2,11 @@ const firefox = require('selenium-webdriver/firefox');
 
 const getFirefoxOptions = function(browser) {
   let firefoxOptions;
-  let path = process.env.KITE_HOME.split("\\").join('/');
-  let profile = path + "/third_party/";
-  if (profile != undefined) {
-    switch(browser.platform) {
+  let kiteHome = process.env.KITE_HOME;
+  if (kiteHome != undefined) {
```
Comment (namvuCosmo) [inline wdmgmt/WebDriverUtility.js]: to compare with undefined, use this `if(typeof obj !== "undefined") {`

File: util/KiteBaseTest.js [hunk 5]
```diff
@@ -33,13 +34,20 @@ class KiteBaseTest {
...
+        this.port = payload.port;
+        let server = 'http://localhost:' + this.port + '/';
+        this.io = io(server);
+      }
       this.timeout = payload.testTimeout;
+      this.takeScreenshot = payload.takeScreenshotForEachTest;
       // getStats info
       let getStats = payload.getStats;
-      if (getStats != undefined) {
+      if (typeof getStats !== "undefined") {
```
Comment (ManuCosmo) [inline util/KiteBaseTest.js]: shouldnt this be simply?
`if (!getStats)`

**Pull request merged**